src/rust/ide/lib/view/src/graph_editor/component/type_coloring.rs

```diff
@@ -0,0 +1,55 @@
+//! This module contains functionality that allows ports and edges to be colored according
+//! to their type information.
+
+use crate::prelude::*;
+
+use crate::graph_editor::SharedHashMap;
+use crate::graph_editor::Type;
+
+use ensogl::data::color;
+use std::collections::hash_map::DefaultHasher;
+use std::hash::Hasher;
+
+
+/// Color that should be used if no type information is available.
+pub const DEFAULT_TYPE_COLOR : color::Lcha = color::Lcha::new(0.5, 0.0, 0.0, 1.0);
+
+
+
+// ================================
+// === Type to Color Conversion ===
+// ================================
+
+fn type_to_hash(type_information:Type) -> u64 {
+    let mut hasher = DefaultHasher::new();
+    type_information.0.hash(&mut hasher);
+    hasher.finish()
+}
+
+/// Return the color that corresponds to the given type. Can be used to color edges and ports.
+pub fn color_for_type(type_information:Type) -> color::Lch {
+    let hue =  (type_to_hash(type_information) % 360) as f32 / 360.0;
```

Comment:
I'm not sure we are using 360 degrees for hue. Arent we using 0-1 there? To be checked and confirmed

Reply:
I've added a note. But essentially, if the hash is random, the 360 here is irrelevant. We normalise to 0..1 in the end.